gateway-ha/src/main/java/io/trino/gateway/ha/router/HaQueryHistoryManager.java

```diff
@@ -77,4 +89,65 @@ public String getBackendForQueryId(String queryId)
         }
         return backend;
     }
+
+    @Override
+    public TableData<QueryDetail> findQueryHistory(QueryHistoryRequest query)
+    {
+        try {
+            connectionManager.open();
+            String sql = "select * from query_history where 1=1";
+            if (StringUtils.isNotBlank(query.getUser())) {
+                sql += " and user_name = '" + query.getUser() + "'";
+            }
+            if (StringUtils.isNotBlank(query.getBackendUrl())) {
+                sql += " and backend_url = '" + query.getBackendUrl() + "'";
+            }
+            if (StringUtils.isNotBlank(query.getQueryId())) {
+                sql += " and query_id = '" + query.getQueryId() + "'";
+            }
+            int start = PageUtil.getStart(query.getPage(), query.getSize());
+            List<QueryDetail> rows = QueryHistory.upcast(QueryHistory.findBySQL(String.join(" ",
+                    sql,
+                    "order by created desc",
+                    "limit ", String.valueOf(start), ",", String.valueOf(query.getSize()))));
+            Long total = QueryHistory.count();
+            return TableData.build(rows, total);
+        }
+        finally {
+            connectionManager.close();
+        }
+    }
+
+    @Override
+    public List<DistributionResponse.LineChart> findDistribution(Long ts)
+    {
+        List<DistributionResponse.LineChart> resList = new ArrayList<>();
+        try {
+            connectionManager.open();
+            String sql = """
+                select FLOOR(created / 1000 / 60)  minute,
+                       backend_url                 ,
+                       count(1)                    query_count
+                from query_history
+                where created > %s
+                group by FLOOR(created / 1000 / 60), backend_url
+                """.formatted(ts);
+            List<Map> results = Base.findAll(sql);
+            for (Map model : results) {
+                DistributionResponse.LineChart lineChart = new DistributionResponse.LineChart();
+                int minute = Integer.parseInt(model.get("minute").toString());
```

Comment:
![image](https://github.com/trinodb/trino-gateway/assets/18670495/4c1ed751-71e1-4048-9825-3f9c3cc31710)

this line returns another error because `model.get("minute")` returns ` 2.8435486E7` that can't be parsed to integer

Reply:
can be fixed with ` long minute = ((Double) model.get("minute")).longValue();`

Reply:
![image](https://github.com/trinodb/trino-gateway/assets/54202482/f1dbe3be-a752-4762-88c1-63b981d164bc)

Our results look different. I think different types of databases may return different results. I used MySQL. What type of database are you?

Reply:
I thought so too. I'm using PostgresSQL

Reply:
I fixed this issue

Reply:
great! Will keep testing it